frontend/views/components/ProfileCard.vue

```diff
@@ -161,28 +162,13 @@ export default ({
     },
     sendMessage () {
       if (!this.ourPrivateDirectMessages[this.username]) {
-        sbp('gi.actions/mailbox/createDirectMessage', {
-          contractID: this.currentIdentityState.attributes.mailbox,
-          data: {
-            privacyLevel: CHATROOM_PRIVACY_LEVEL.PRIVATE,
-            usernames: [this.username]
-          }
-        })
+        this.createPrivateDM(this.username)
       } else {
         const chatRoomId = this.directMessageIDFromUsername(this.username)
         if (this.ourPrivateDirectMessages[this.username].hidden) {
-          sbp('gi.actions/mailbox/setDirectMessageVisibility', {
-            contractID: this.currentIdentityState.attributes.mailbox,
-            data: {
-              contractID: chatRoomId,
-              hidden: false
-            }
-          })
+          this.setDMVisibility(chatRoomId, false)
         } else {
-          this.$router.push({
-            name: 'GroupChatConversation',
-            params: { chatRoomId }
-          }).catch(logExceptNavigationDuplicated)
+          this.redirect(chatRoomId)
```

Comment:
None of the potential errors are being handled here, so I opened a separate issue to address this: #1647